Sources/CocoaLumberjackSwift/DDLog+Combine.swift

```diff
@@ -0,0 +1,122 @@
+// Software License Agreement (BSD License)
+//
+// Copyright (c) 2010-2020, Deusty, LLC
+// All rights reserved.
+//
+// Redistribution and use of this software in source and binary forms,
+// with or without modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice,
+//   this list of conditions and the following disclaimer.
+//
+// * Neither the name of Deusty nor the names of its contributors may be used
+//   to endorse or promote products derived from this software without specific
+//   prior written permission of Deusty, LLC.
+
+#if canImport(Combine)
+
+import Combine
+
+#if SWIFT_PACKAGE
+import CocoaLumberjack
+import CocoaLumberjackSwiftSupport
+#endif
+
+@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+extension DDLog {
+
+    /**
+     * Creates a message publisher.
+     *
+     * The publisher will add and remove loggers as subscriptions are added and removed.
+     *
+     * The level that you provide here is a preemptive filter (for performance).
+     * That is, the level specified here will be used to filter out logMessages so that
+     * the logger is never even invoked for the messages.
+     *
+     * More information:
+     *    See -[DDLog addLogger:with:]
+     *
+     * - Parameter logLevel: preemptive filter of the message returned by the publisher. All levels are sent by default
+     * - Returns: A MessagePublisher that emits LogMessages filtered by the specified logLevel
+     **/
+    public func messagePublisher(with logLevel: DDLogLevel = .all) -> MessagePublisher {
+        return MessagePublisher(log: self, with: logLevel)
+    }
+
+    // MARK: - MessagePublisher
+    
+    @available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+    public struct MessagePublisher: Combine.Publisher {
+
+        public typealias Output = DDLogMessage
+        public typealias Failure = Never
+
+        private let log: DDLog
+        private let logLevel: DDLogLevel
+
+        public init(log: DDLog, with logLevel: DDLogLevel) {
+            self.log = log
+            self.logLevel = logLevel
+        }
+
+        public func receive<S>(subscriber: S) where S: Subscriber, S.Failure == Never, S.Input == DDLogMessage {
```

Comment:
NitPick: we should use the type aliases for the generic restrictions here:
```suggestion
        public func receive<S>(subscriber: S) where S: Subscriber, S.Input == Output, S.Failure == Failure {
```

Reply:
Done